src/RepositoryChainsArchiver.ts

```diff
@@ -51,105 +63,107 @@ export default class RepositoryChainsArchiver {
 
-      // For each entry, add the file to the appropriate tar stream
-      for await (const entry of entries) {
-        const relativePath = entry.toString();
-        const filePath = path.join(this.repositoryPath, relativePath);
-
-        // Extract matchType and other necessary information from the relative path
-        const pathParts = relativePath.split("/");
-        const matchType = pathParts[0]; // 'full_match' or 'partial_match'
-        const chainId = pathParts[1]; // 'full_match' or 'partial_match'
-        const address = pathParts[2]; // Assuming path format: 'matchType/chainId/addressDirName/...'
-
-        if (!address.startsWith("0x")) {
-          console.log("Skipping non-address folder:", filePath);
-          continue;
-        }
-
-        if (matchType !== "full_match" && matchType !== "partial_match") {
-          console.log("Skipping non-match folder:", filePath);
-          continue;
+    try {
+      const uploadedFiles: { path: string; sizeInBytes: number }[] = [];
+      for (const matchType of ["full_match", "partial_match"]) {
+        for (const chainId of this.chainIds) {
+          for (let i = 0; i < 256; i++) {
+            const byte = i.toString(16).padStart(2, "0");
+            console.log(`Processing ${matchType} ${chainId} ${byte}`);
+            const pattern = `${matchType}/${chainId}/0x${byte}*/**/*`;
+            const entries = fg.stream(pattern, {
+              cwd: baseDir,
+              onlyFiles: true,
+              concurrency: READ_CONCURRENCY,
+              caseSensitiveMatch: false,
+            });
+
+            // Create tar stream for current byte
+            const archiveName = `${matchType}.${chainId}.${byte}.tar.gz`;
+            const localPath = path.join(this.exportPath, archiveName);
+            let currentTarStream: tar.Pack | undefined;
+
+            for await (const entry of entries) {
+              if (!currentTarStream) {
+                currentTarStream = createPack(localPath);
+              }
+
+              const relativePath = entry.toString();
+              const filePath = path.join(this.repositoryPath, relativePath);
+
+              const pathParts = relativePath.split("/");
+              const address = pathParts[2]; // Assuming path format: 'matchType/chainId/addressDirName/...'
+
+              if (!address.startsWith("0x")) {
+                console.log("Skipping non-address folder:", filePath);
+                continue;
+              }
+
+              await this.addFileToPack(currentTarStream, filePath);
+            }
+
+            if (currentTarStream) {
+              currentTarStream.finalize();
+
+              // Upload file
+              const s3Key = `${backupName}/${archiveName}`;
+              await this.uploadFile(localPath, s3Key);
+              uploadedFiles.push({
+                path: s3Key,
+                sizeInBytes: fs.statSync(localPath).size,
+              });
```

Comment:
Yeah, might be worth to add some benchmarking. Let's see how it performs now. I also increased the Cloud Run CPU and memory a bit. I don't expect the upload to take too long, because the archives are in the size of around 12 MB.